innkeeper/client.go

```diff
@@ -320,6 +318,43 @@ func getHttpError(r *http.Response) (error, bool) {
 	}
 }
 
+func (c *Client) writeRoute(url string, route *routeData) error {
+
+	res, err := json.Marshal(route)
+
+	req, err := http.NewRequest("POST", url, bytes.NewReader(res))
+
+	if err != nil {
+		return err
+	}
+
+	authToken, err := c.opts.Authentication.GetToken()
```

Comment:
I agree with dan. I would not add caching of tokens here. If the thing, that provides the token would have to make an expensive call, it is the responsibility of that thing to cache.